python/plugins/processing/algs/qgis/ui/ExecuteSQLWidget.py

```diff
@@ -95,6 +95,17 @@ def setValue(self, value):
                             and v.source() == QgsProcessingModelChildParameterSource.ExpressionText:
                         text = v.expressionText()
 
+                        # replace parameter's name by expression (diverging after model save)
+                        names = QgsExpression.referencedVariables(text)
+
+                        strings = self.dialog.getAvailableValuesOfType(
+                            [QgsProcessingParameterString, QgsProcessingParameterNumber], [])
+                        model_params = [(self.dialog.resolveValueDescription(s), s) for s in strings]
+
+                        for k, v in model_params:
+                            if v.parameterName() in names:
+                                text = text.replace('[% @{} %]'.format(v.parameterName()), '[% @{} %]'.format(k))
```

Comment:
There is some spacing here between the percent and the variable which has been removed above. Is this done deliberately?

Reply:
> Is this done deliberately?

Yes it is because the `QgsFieldExpressionWidget` already adds space before/after the expression.